### PR TITLE
Feature/strike widget

### PR DESCRIPTION
**Pull request by leonkho**
- Added widget to show how close current price is to strike price
- Includes liquidity gauge staking APY
  - Shows ⚡️ indicator for vaults with staking enabled
- Updated APY to use the average between last 4 rounds (or weeks)
  - We don't always get 4 rounds. Some new vaults don't have 4 rounds, then we just average however many rounds we have
  - Only account for positive weeks (so if 1/4 rounds is negative, we will only get the average between the 3 positive rounds)

New APY for vaults + new strike widget
Screenshot 2022-04-10 at 7 48 27 PM

How strike widget looks when ITM (This is a put vault)
Screenshot 2022-04-10 at 7 35 55 PM

### CODE REVIEW

**Comment by leonkho** [comment]
@cjinghong, this commit removes all chainID checks from vault version function:
https://github.com/ribbon-finance/ribbon-frontend/commit/39559876bd246fe8bb3a2503d243b58722e58dd9

File: shared/src/hooks/useLatestAPY.ts [hunk 1]
```diff
@@ -107,6 +113,7 @@ export const calculateAPYFromPriceHistory = (
   }
 
   let weeksAgo = 0;
+  const apys: number[] = [];
```
Comment (cjinghong) [inline shared/src/hooks/useLatestAPY.ts:116]: Here we create an array to keep track of the APY calculation between each rounds

File: shared/src/hooks/useLatestAPY.ts [hunk 2]
```diff
@@ -124,74 +131,77 @@ export const calculateAPYFromPriceHistory = (
         ),
       ];
 
-      /**
-       * If the given period is profitable, we calculate APY
-       */
+      // Only includes weeks that is positive
       if (endingPricePerShare > startingPricePerShare) {
```
Comment (cjinghong) [inline shared/src/hooks/useLatestAPY.ts:135]: We are still ONLY including weeks that are positive

File: shared/src/hooks/useLatestAPY.ts [hunk 3]
```diff
@@ -124,74 +131,77 @@ export const calculateAPYFromPriceHistory = (
...
+    // Continue to previous week
     periodStart.subtract(1, "week");
     weeksAgo += 1;
 
     /**
      * Prevent searching too far ago
      */
-    if (weeksAgo >= 5) {
-      return 0;
+    if (weeksAgo >= 4) {
```
Comment (cjinghong) [inline shared/src/hooks/useLatestAPY.ts:198]: We break the loop when we have past 4 weeks APY

File: shared/src/hooks/useLatestAPY.ts [hunk 4]
```diff
@@ -124,74 +131,77 @@ export const calculateAPYFromPriceHistory = (
...
+      break;
     }
   }
+
+  // Calculate average apy
+  return apys.reduce((acc, curr) => acc + curr, 0) / apys.length || 0;
```
Comment (cjinghong) [inline shared/src/hooks/useLatestAPY.ts:204]: Get the average here

**Review by cjinghong — APPROVED** [review]

**Pull request merged**